Mathlib/Algebra/Order/BigOperators/GroupWithZero/List.lean

```diff
@@ -58,6 +58,23 @@ theorem prod_map_le_prod_map₀ {ι : Type*} {s : List ι} (f : ι → R) (g : 
       simp [ha]
     · apply (h0 _ _).trans (h _ _) <;> simp
 
+theorem prod_map_le_pow_length {F L : Type*} [MulPosMono R] [FunLike F L R] {f : F} {r : R}
+    {t : List L} (hf0 : ∀ x ∈ t, 0 ≤ f x) (hf : ∀ x ∈ t, f x ≤ r) :
+    (List.map f t).prod ≤ r ^ length t := by
+  induction t with
+  | nil => simp
+  | cons a s ih =>
+    simp only [map_cons, prod_cons, length_cons, length_map, pow_succ']
+    have har : f a ≤ r := by
+      apply hf
+      simp
+    gcongr
+    · apply prod_nonneg (fun _ ha ↦ ?_)
+      obtain ⟨j, hj, rfl⟩ := mem_map.mp ha
+      exact hf0 j (mem_cons_of_mem a hj)
+    · exact le_trans (hf0 a mem_cons_self) har
+    · simp_all
+
```

Comment:
```suggestion
 induction t with
  | nil => simp
  | cons a s ih =>
    simp only [map_cons, prod_cons, length_cons, length_map, pow_succ']
    gcongr
    · apply prod_nonneg (fun _ ha ↦ ?_)
      obtain ⟨j, hj, rfl⟩ := mem_map.mp ha
      exact hf0 j (mem_cons_of_mem a hj)
    · exact le_trans (hf0 a mem_cons_self) <| hf _ (by simp)
    · simp_all
```

Reply:
There's one goal still open if I apply your suggestion; did you not copy the complete suggested proof by mistake?

Reply:
No, I tried again and it does not work. Forget my suggestion, sorry for the noise.